src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DELIMITER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String SUPPLY_TYPE = "supply";
+    private static final String BUY_TYPE = "buy";
+    private static final String RESULT_TYPE = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readDataFromFile(fromFileName);
+        String report = createReport(data);
+        writeDataToFile(toFileName, report);
+    }
+
+    private String[] readDataFromFile(String filename) {
+        StringBuilder supplyBuyInfo = new StringBuilder();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filename))) {
+            String line;
+
+            while ((line = bufferedReader.readLine()) != null) {
```

Comment:
You can create the inner variable 
String line = bufferedReader.readLine() 
to avoid creation in the while block and make your code more readable